src/shared/i18n/I18NFrench.cpp

```diff
@@ -18,441 +18,441 @@
+    { I18N_MAINMENU_SERVER_NAME, "Nom du serveur" },
+    { I18N_MAINMENU_CONNECT, "Se connecter" },
+    { I18N_MAINMENU_REFRESH, "Rafraichir" },
+    { I18N_MAINMENU_PLAYERS, "Joueurs" },
     { I18N_MAINMENU_PING, "Ping" },
-    { I18N_MAINMENU_LAN_TITLE, "LAN" },
-    { I18N_MAINMENU_LAN_SERVERS, "LAN servers:" },
-    { I18N_MAINMENU_CUSTOM_SERVER, "Custom Server" },
-    { I18N_MAINMENU_ENTER_HOSTNAME, "Enter here the hostname or IP address and port:" },
-    { I18N_MAINMENU_ADDRESS, "address:" },
-    { I18N_MAINMENU_PORT, "port:" },
-    { I18N_MAINMENU_PASSWORD, "password:" },
-    { I18N_MAINMENU_ENTER_PASSWORD, "Enter password:" },
-    { I18N_MAINMENU_SERVER_NAME2, "server name:" },
-    { I18N_MAINMENU_GAME_MODE, "game mode:" },
-    { I18N_MAINMENU_GM_DM, "death match" },
-    { I18N_MAINMENU_GM_TDM, "team death match" },
-    { I18N_MAINMENU_GM_CTF, "capture the flag" },
-    { I18N_MAINMENU_GM_SR, "speed race" },
-    { I18N_MAINMENU_GM_CTC, "catch the coin" },
-    { I18N_MAINMENU_GM_GOH, "goat of the hill" },
-    { I18N_MAINMENU_MAX_PLAYERS, "max. players:" },
-    { I18N_MAINMENU_WARMUP, "warm-up:" },
-    { I18N_MAINMENU_DURATION, "duration:" },
-    { I18N_MAINMENU_IN_SECONDS, "(in seconds)" },
-    { I18N_MAINMENU_IN_MINUTES, "(in minutes)" },
-    { I18N_MAINMENU_SELECT_MAP, "select map:" },
-    { I18N_MAINMENU_INVALID_SERVERNAME, "Please enter a valid server name." },
-    { I18N_MAINMENU_MAX_PLAYERS_TITLE, "Max. Players" },
-    { I18N_MAINMENU_INVALID_GENERIC_NUMBER, "Please enter a valid number. (${1})" },
-    { I18N_MAINMENU_DURATION2, "Duration" },
-    { I18N_MAINMENU_WARMUP2, "Warm-Up" },
-    { I18N_MAINMENU_SELECT_MAP2, "Select Map" },
-    { I18N_MAINMENU_SELECT_MAP3, "Please select a map." },
-    { I18N_MAINMENU_PACKAGE, "Package" },
-    { I18N_MAINMENU_HASH, "Hash" },
-    { I18N_MAINMENU_CREDITS_CODE, "code:" },
-    { I18N_MAINMENU_CREDITS_GFX, "graphics:" },
-    { I18N_MAINMENU_CREDITS_MAPS, "maps:" },
-    { I18N_MAINMENU_CREDITS_MUSIC, "music:" },
-    { I18N_MAINMENU_CREDITS_THANKS, "special thanks to:" },
-    { I18N_MAINMENU_FREDERIC, "and of course, frederic the goat" },
+    { I18N_MAINMENU_LAN_TITLE, "Local" },
+    { I18N_MAINMENU_LAN_SERVERS, "Serveurs locaux:" },
+    { I18N_MAINMENU_CUSTOM_SERVER, "Serveur personalisé" },
+    { I18N_MAINMENU_ENTER_HOSTNAME, "Entrez ici l'adresse IP et le port:" }, 
+    { I18N_MAINMENU_ADDRESS, "Adresse:" },
+    { I18N_MAINMENU_PORT, "Port:" },
+    { I18N_MAINMENU_PASSWORD, "Mot de passe:" },
+    { I18N_MAINMENU_ENTER_PASSWORD, "Entrer le mot de passe:" },
+    { I18N_MAINMENU_SERVER_NAME2, "Nom du serveur:" },
+    { I18N_MAINMENU_GAME_MODE, "Mode de jeu:" },
+    { I18N_MAINMENU_GM_DM, "Match à mort" },
+    { I18N_MAINMENU_GM_TDM, "Match à mort par équipe" },
+    { I18N_MAINMENU_GM_CTF, "Capture de drapeau" },
+    { I18N_MAINMENU_GM_SR, "Course" },
+    { I18N_MAINMENU_GM_CTC, "Possession" },
```

Comment:
Not sure about translating "Catch the coin" this way. It mostly depends on what the game mode is exactly and how it should be branded. More literal translations could be something like "Chasse aux pièces" or "Attrappez les pièces". Would need to actually try that game mode to get a better opinion.